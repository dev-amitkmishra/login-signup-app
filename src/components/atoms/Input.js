import React from 'react';
import PropTypes from 'prop-types';

const Input = ({
    name,
    changeHandler,
    autocompleteText,
    isRequired,
    blurHandler,
    type,
    value
}) => {
    return <input type={type} value={value} name={name} onChange={changeHandler} onBlur={blurHandler} autoComplete={autocompleteText} required={isRequired} />
}

Input.propTypes = {
    name: PropTypes.string.isRequired,
    changeHandler: PropTypes.func,
    autocompleteText: PropTypes.string,
    isRequired: PropTypes.bool,
    blurHandler: PropTypes.func,
    type: 'text',
    value: ''
}

Input.defaultProps = {
    name: '',
    changeHandler: () => {},
    autocompleteText: '',
    isRequired: true,
    blurHandler:  () => {}
}

export default Input;