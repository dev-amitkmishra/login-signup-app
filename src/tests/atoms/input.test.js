import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../../components/atoms/Input';

describe('Input', () => {
    it('renders correctly', () => {
        const mockfn = jest.fn();
        const tree = renderer.create(<Input classes='test' handler={mockfn} text='test' />).toJSON();
        expect(tree).toMatchSnapshot();
    });
})