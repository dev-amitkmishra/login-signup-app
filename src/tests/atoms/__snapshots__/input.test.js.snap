// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<input
  autoComplete=""
  name=""
  onBlur={[Function]}
  onChange={[Function]}
  required={true}
/>
`;
